dropshot/src/handler.rs

```diff
@@ -72,13 +72,18 @@ pub type HttpHandlerResult = Result<Response<Body>, HttpError>;
 
 /// Handle for various interfaces useful during request processing.
 #[derive(Debug)]
+#[non_exhaustive]
 pub struct RequestContext<Context: ServerContext> {
     /// shared server state
     pub server: Arc<DropshotState<Context>>,
     /// HTTP request routing variables
     pub path_variables: VariableSet,
     /// expected request body mime type
     pub body_content_type: ApiEndpointBodyContentType,
+    /// Maximum request body size: typically the same as
+    /// [`server.config.request_body_max_bytes`], but can be overridden for an
+    /// individual request
+    pub request_body_max_bytes: usize,
```

Comment:
What do you think about making this a method instead?  This feels pretty similar to `RequestContext::page_limit()`.  In both cases we want to provide the "appropriate" value for some tunable based on some non-trivial policy related to multiple places it might be set.

Reply:
So I think we still need to store the value in the `RequestContext` somewhere -- what about storing it in `RequestInfo`? Then this could be a method `rqctx.request.body_max_bytes()`, or `rqctx.request.request_body_max_bytes()`.

Or do you have a way to avoid storing the value on the request? I couldn't find an easy answer to this.

Reply:
Sorry, I should have been clearer -- I'm fine with storing it there.  I'm suggesting it not be `pub` and instead provide a method to access it (that takes into account the various caps).